#params from adaprive clustering 
/clustering_node:
  ros__parameters:
    input_topic: /lidar_points #"/patchworkpp/nonground"
    frame_id: "hesai_lidar"

    #clustering
    minClusterSize : 1        # Minimum cluster size to filter out noise
    maxClusterSize : 1000        # Maximum size for large objects
    voxelX: 0.4               #// Down-sampling resolution in X (meters)
    voxelY: 0.4                  #// Down-sampling resolution in Y (meters)
    voxelZ: 0.4                #// Down-sampling resolution in Z (meters)
    countThreshold: 1          #  // Minimum points per voxel

    #filter for clusters
    clusterMaxX: 0.3
    clusterMaxY: 0.3
    clusterMaxZ: 0.4
    clusterMinX: 0.0
    clusterMinY: 0.0
    clusterMinZ: 0.0
    maxHeight: 1.0          #of the lowest point

    #filter on Z
    downFilterLimits: 0.0
    upFilterLimits: 0.7

    #segmentation
    distanceThreshold: 0.5
    maxIterations: 500
    probability: 0.90
    optimizeCoefficients: true

    #options
    filterOnZ: true
    segment: false
    publishFilteredPc: true
    publishSegmentedPc: true 
    
    

